### PR TITLE
Stop non-admin ArgoCD users from creating applications via UI

### PR DESCRIPTION
**Pull request by anishasthana**
## This introduces a breaking change

- [x] Yes
- [ ] No

## This Pull Request implements

This will force users to utilize git to create new ArgoCD-managed applications

### CODE REVIEW

**Comment by HumairAK** [comment]
going to hold on this until we have argocd managing argocd 
/hold

File: vars/prod-vars.yaml [hunk 1]
```diff
@@ -599,8 +598,17 @@ argo_rbac_cm: |
       g, aicoe-thoth-devops, role:argocd-admin
       g, aicoe-thoth-devops, role:thoth-admin
```
Comment (HumairAK) [inline vars/prod-vars.yaml:599]: these can be removed since you assign them later

**Comment by goern** [comment]
What do these changes mean? why do we get an extra project for thoth-dev? :)

**Comment by anishasthana** [comment]
@goern Basically this would result in the thoth team having the following two projects:
1) `thoth-dev`:Thoth admins would retain the ability to create applications directly via the argocd ui without having to go through git
2) `thoth`: Thoth-admins will _not_ be able to create applications directly via the ui, so all applications in this project would live completely in git.

This way we could maintain a distinction between test and non-test environments while also permitting users to develop rapidly.

**Pull request merged**